lib/mastodon/snowflake.rb

```diff
@@ -114,6 +137,13 @@ def ensure_id_sequences_exist
       end
     end
 
+    def id_at(timestamp)
+      id  = timestamp.to_i * 1000
```

Comment:
Note that this forces the millisecond part to zero. If we'll never get one from a source where this would be called, this is vaguely OK (though I'd prefer to add them in if they exist), but we might also want to just pick a random millisecond if it's zero? Lower likelihood of hitting a duplicate ID, not that it's all that likely in the first place.

Reply:
I figure since we only ever call this from networked timestamps, usec would always be 0, therefore it can be skipped. I agree we could do a random millisecond part though. ...How exactly do I do that? `+rand(1000)`?

Reply:
Yep, I kind of figured that was why, and it doesn't bother me too much.

Just adding a `+ rand(1000)` should do it, indeed.